examples/dataflow-flex-python/df-sample-xml-to-bq/xmltobq.py

```diff
@@ -0,0 +1,128 @@
+import argparse
+import logging
+
+import apache_beam as beam
+import apache_beam.io.fileio as fileio
+
+table_schema = {
+    'fields': [
+        {
+            'name': 'CustomerID',
+            'type': 'STRING',
+            'mode': 'NULLABLE'
+        },
+        {
+            'name': 'EmployeeID',
+            'type': 'STRING',
+            'mode': 'NULLABLE'
+        },
+        {
+            'name': 'OrderDate',
+            'type': 'STRING',
+            'mode': 'NULLABLE'
+        },
+        {
+            'name': 'RequiredDate',
+            'type': 'STRING',
+            'mode': 'NULLABLE'
+        },
+        {
+            'name':
+                'ShipInfo',
+            'type':
+                'RECORD',
+            'mode':
+                'NULLABLE',
+            'fields': [
+                {
+                    'name': 'ShipVia',
+                    'type': 'STRING',
+                    'mode': 'NULLABLE'
+                },
+                {
+                    'name': 'Freight',
+                    'type': 'STRING',
+                    'mode': 'NULLABLE'
+                },
+                {
+                    'name': 'ShipName',
+                    'type': 'STRING',
+                    'mode': 'NULLABLE'
+                },
+                {
+                    'name': 'ShipAddress',
+                    'type': 'STRING',
+                    'mode': 'NULLABLE'
+                },
+                {
+                    'name': 'ShipCity',
+                    'type': 'STRING',
+                    'mode': 'NULLABLE'
+                },
+                {
+                    'name': 'ShipRegion',
+                    'type': 'STRING',
+                    'mode': 'NULLABLE'
+                },
+                {
+                    'name': 'ShipPostalCode',
+                    'type': 'STRING',
+                    'mode': 'NULLABLE'
+                },
+                {
+                    'name': 'ShipCountry',
+                    'type': 'STRING',
+                    'mode': 'NULLABLE'
+                },
+            ]
+        },
+    ]
+}
+
+
+def parse_to_dict(file):
+    import xmltodict
+
+    parsed_xml = xmltodict.parse(file.read())
+    return [parsed_xml]
+
+
+def get_orders(doc):
+    for order in doc['Root']['Orders']['Order']:
+        yield order
```

Comment:
This works for the example provided, but is quite fragile if Orders is empty (which is allowed according to the XSD), since xmltodict would set its value to `None`. Additionally, as mentioned above, if Orders were to define its elements as some base type and we were to specialize into multiple order types like PhoneOrder, MailOrder, StoreOrder, InternalOrder, etc. then Orders would have a property for each type. Again, this is due to how xmltodict converts the file though. A data binding library would produce output that's more structured and easy to use.

Additionally, you could use `yield from` instead of iterating over the elements and yielding them manually.

Reply:
Added Yield and using xsdata now.